### PR TITLE
Mission Bugs: Fixing Pausing Bugs in Cutscenes + Others

### PR DESCRIPTION
**Pull request by jurcaua**
* reset birdies velocity on restart of mission
* move pause menu above all UI
* update cutscene animations according to their updateMode now
* fix bug that ended tut mission more than once in some cases

### CODE REVIEW

File: Assets/Scripts/Missions/MissionKitchen1.cs [hunk 1]
```diff
@@ -280,10 +280,12 @@ private void OnFadingCompleteRestartMission()
             }
 
             GameManager.Instance.GetPlayerTransform().position = denturesCheckpointRespawnPosition;
+            GameManager.Instance.GetMovementController().ResetVelocity();
         }
         else
         {
             GameManager.Instance.GetPlayerTransform().position = respawnPosition;
+            GameManager.Instance.GetMovementController().ResetVelocity();
```
Comment (chadleytan) [inline Assets/Scripts/Missions/MissionKitchen1.cs]: Can't we just remove lines 283 and 288 to outside the if-else conditionals or does it need to be reset before you do the `FadeIn`?

File: Assets/Scripts/Missions/MissionKitchen1.cs [hunk 2]
```diff
@@ -280,10 +280,12 @@ private void OnFadingCompleteRestartMission()
...
             UIManager.Instance.FadeIn();
             isRestarting = false;
         }
```
Comment (chadleytan) [inline Assets/Scripts/Missions/MissionKitchen1.cs]:
```suggestion
            }

            GameManager.Instance.GetPlayerTransform().position = denturesCheckpointRespawnPosition;
        }
        else
        {
            GameManager.Instance.GetPlayerTransform().position = respawnPosition;
            UIManager.Instance.FadeIn();
            isRestarting = false;
        }
        GameManager.Instance.GetMovementController().ResetVelocity();
```

Comment (chadleytan) [reply Assets/Scripts/Missions/MissionKitchen1.cs]: Something like this?

**Review by chadleytan — COMMENTED** [review]
Looks good for the most part other than that one thangg

File: Assets/Scripts/Missions/MissionKitchen1.cs (hunk 1 above)
Comment (jurcaua) [reply Assets/Scripts/Missions/MissionKitchen1.cs]: True! I think resetting it before / after doesn't have a noticeable effect at all, so should be good!

**Review by chadleytan — APPROVED** [review]
Sweeeeeetazzzz

**Pull request merged**